### DISCUSSION TITLE
Collection Tree - Get old and new entries/urls after saving

### DISCUSSION THREAD

**Discussion by wuifdesign** [statamic/cms#10618] [Q&A]
Is there a way to geht the old and new paths of moved entries in a collection tree?

i want to create redirects for all affected entries after a node was moved.

**Comment by edalzell** [comment]
There's an addon on that does this, called Redirect, give that a try?

**Reply by wuifdesign** [reply]
this addon also only does it for slug changes, but not moves inside the collection: 
https://github.com/riasvdv/statamic-redirect/blob/main/src/RedirectServiceProvider.php#L61

as `EntrySaved` isn't triggered on collectiontree changes it does not work.

**Reply by edalzell** [reply]
ah ok, so you could PR that into that addon, or learn how it does it and make it yourself, if their license allows you to copy code. It's mostly listeners to the correct events

**Reply by wuifdesign** [reply]
https://github.com/riasvdv/statamic-redirect/issues/93 there is already an issue, but i think with the current code of statamic this is not possible. not without https://github.com/statamic/ideas/issues/1207. As it isn't possible to get the old and the new paths of entries wenn saving a collection tree (like it is possible for entries, as the have the `EntrySaving` event).

but i still wanted to ask if someone knows a workaround.

**Reply by edalzell** [reply]
> but i still wanted to ask if someone knows a workaround.

I'd PR them into Statamic, then patch it in, then use them to build what you want.

**Comment by ryanmitchell** [comment]
@wuifdesign https://github.com/statamic/cms/pull/10625

**Reply by wuifdesign** [reply]
thx, looks nice but misses some code, i commented in the PR
